hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnection.java

```diff
@@ -72,32 +68,35 @@ public class TcpClientConnection implements ClientConnection {
     private final long startTime = System.currentTimeMillis();
     private final Consumer<ClientMessage> responseHandler;
     private final ConcurrentMap attributeMap;
-
+    private final AltoConnectionContext altoConnectionContext;
     @Probe(name = CLIENT_METRIC_CONNECTION_EVENT_HANDLER_COUNT, level = MANDATORY)
     private final ConcurrentMap<Long, EventHandler> eventHandlerMap = new ConcurrentHashMap<>();
-
-    private volatile Address remoteAddress;
     @Probe(name = CLIENT_METRIC_CONNECTION_CLOSED_TIME, level = ProbeLevel.DEBUG)
     private final AtomicLong closedTime = new AtomicLong();
 
+    private volatile Address remoteAddress;
     private volatile Throwable closeCause;
     private volatile String closeReason;
-    private String connectedServerVersion;
     private volatile UUID remoteUuid;
     private volatile UUID clusterUuid;
-    private volatile Channel[] tpcChannels;
-    public volatile ChannelInitializer channelInitializer;
 
-    public TcpClientConnection(HazelcastClientInstanceImpl client, int connectionId, Channel channel) {
+    // All updates to this apart from the initial setter must be made
+    // guarded by the connection itself.
+    private volatile TcpClientAltoConnection[] altoConnections;
```

Comment:
Why do we need to wrap the tpcChannels into connections? I don't see the added value.

Reply:
It doesn't add any value as you said, but It was necessary if we want to distinguish invocations sent to classic connections and Alto connections in the invocation service. So, if an Alto connection is closed, we would only notify invocations sent to it with TargetDisconnectedException. 

But, you are right as this adds too much unnecessary code. Let me try to find a way to simplify it